### DISCUSSION TITLE
Run task via API with enviroment variable get 500

### DISCUSSION THREAD

**Discussion by dmitrygribenchuk** [semaphoreui/semaphore#1044] [Q&A] [answered]
Hello. I try to run a task via API. Without an environment variable all work fine. And with a specified environment (found it in API docs). I created unbound environment in UI. I'm get a 500 error.

Endpoint: http://localhost:3000api/project/1/tasks

body: 

{
	"template_id": 2,
	"status": "running",
        "environment": "unbound"
}

In logs:
time="2022-10-06T09:30:25Z" level=error msg="invalid character 'u' looking for beginning of value" error="Cannot write new event to database"

**Comment by fiftin** [comment]
Hi @dmitrygribenchuk ,

`environment` must be JSON, it is not name of existing environment.

I agree error message must be more informative.

**Comment by dmitrygribenchuk** [comment]
Thanks for your answer, on a docs environment must be string
https://www.ansible-semaphore.com/api-docs/#/project/post_project__project_id__tasks

{
  "template_id": 0,
  "debug": true,
  "dry_run": true,
  "playbook": "string",
  "environment": "string"
}

**Comment by fiftin** [comment, marked as answer]
Yes, it should be stringified JSON.
```
"environment": "{\"prop\": \"value\"}"
```

**Comment by dmitrygribenchuk** [comment]
Thanks for your reply! It's works.
